src/machines/java/com/enderio/machines/common/blockentity/AlloySmelterBlockEntity.java

```diff
@@ -191,58 +195,68 @@ public AlloySmeltingMachineTask(@NotNull Level level, MachineInventory inventory
             super(level, inventory, energyStorage, container, outputSlot, recipe);
         }
 
+        @Override
+        protected void onDetermineOutputs(AlloySmeltingRecipe recipe) {
+            // This handles the output multiplication for vanilla smelting recipes.
+            if (recipe instanceof VanillaAlloySmeltingRecipe) {
+                CountedIngredient input = recipe.getInputs().get(0);
+
+                int inputCount = 0;
+                for (SingleSlotAccess slot : INPUTS.getAccesses()) {
+                    if (input.test(slot.getItemStack(getInventory()))) {
+                        inputCount++;
+                    }
+                }
```

Comment:
make it so that it can extract up to 3 recipes from the same slot. Could you also reverse the lookup for this, so that it takes the last slot first?

Reply:
That might be a nice QoL change in terms of taking multiple from the same slot.
Only problem is this could tie into #251 and im unsure on how to try tackle that one. I guess a custom container wrapper that'll check all three slots for index 0?

Reply:
that's right, we can merge this then and I'll tackle 251 next